tests/aws/services/lambda_/functions/common/echo/dotnet/Makefile

```diff
@@ -0,0 +1,33 @@
+UNAME := $(shell uname -m)
+ifeq ($(UNAME),x86_64)
+	ARCHITECTURE ?= x86_64
+else
+	ARCHITECTURE ?= arm64
+endif
+# The packaging function architecture is x86_64 by default and needs to be set explicitly for arm64
+# https://github.com/aws/aws-extensions-for-dotnet-cli/blob/cdd490450e0407139d49248d94a4a899367e84df/src/Amazon.Lambda.Tools/LambdaDefinedCommandOptions.cs#L111
+FUNCTION_ARCHITECTURE ?= $(ARCHITECTURE)
+
+# Target Dotnet framework version
+FRAMEWORK ?= net8.0
+
+# https://gallery.ecr.aws/sam/build-dotnet8
+IMAGE ?= public.ecr.aws/sam/build-dotnet8:1.112.0
+
+# Emulated builds with Dotnet8 are currently (2024-03-19) broken as discussed in many issues:
+# https://github.com/NuGet/Home/issues/12227
+# https://github.com/dotnet/runtime/issues/78340
+# https://github.com/dotnet/msbuild/issues/8508
+# Root cause QEMU issue: https://gitlab.com/qemu-project/qemu/-/issues/249
+# Workaround: Instead of emulating the build (works for Dotnet6), we use the native Docker image
+#		      and cross-build the Dotnet package using the flag `--function-architecture` (x86_64 or arm64).
+
+build:
+	mkdir -p build && \
+	docker run --rm -v $$(pwd)/src:/app -v $$(pwd)/build:/out $(IMAGE) bash -c "mkdir -p /app2 && cp /app/* /app2 && cd /app2 && dotnet lambda package --framework $(FRAMEWORK) --function-architecture $(FUNCTION_ARCHITECTURE) -o ../out/handler.zip" && \
```

Comment:
There is currently a flaky issue when building two Dotnet runtimes in short succession:

```
cd localstack/tests/aws/services/lambda_/functions/common/echo
# FAILS
make -C dotnet build && make -C dotnet6 build
# WORKS
make -C dotnet build && sleep 1 && make -C dotnet6 build
```

Error:
```
Zipping publish folder /app2/bin/Release/net6.0/publish to /out/handler.zip
... zipping: zip I/O error: No such file or directory
... zipping: zip error: Could not create output file (/out/handler.zip)
Lambda project successfully packaged: /out/handler.zip
cp: build/handler.zip: No such file or directory
make[2]: *** [build] Error 1
make[1]: *** [build] Error 2
make: *** [dotnet6] Error 2
```

Maybe that's related to the Docker container, which needs some time to shut down 🤔 
@dfangl Do you have any suggestions on how we should guard against this to avoid a flaky build and fix it locally

Reply:
Very possible that there is a race condition here. Maybe we could avoid having the same mount for both? Passing in the build folder so we have a different one for dotnet6 and dotnet8 builds?

Reply:
Good idea 👍 
I fixed it using the `FRAMEWORK` version as a natural build directory suffix to minimize sub-Makefile changes.